mantidproject/mantid
# Move validators for indirect/inelastic interfaces to common library

### Description of work
This interface moves input validators used in indirect/inelastic interfaces from several redundant definitions in anonymous namespaces to a common file, in this case the `InterfaceUtils` library. Additionally, other repeated functions have been moved as well. In order to provide access to some classes using the validators (like workspace dialogs), the `InterfaceUtils` and `WorkspaceUtils` files have been moved from the `CustomInterfaces` to the `MantidWidgets` library. 
#### Summary of work




Fixes #37009  




#### Further detail of work


### To test:
- On `Interfaces ->Data Analysis`, click on `Add Workspace` on either of the three first tabs and check that the `Workspace Indices` line edit only accepts integer numerical input separated by commas or dashes: `0,1,2-3`, `0-3`, `0,1`
- On `Interfaces -> Data Analysis`, add data on any of the tabs and check that the `Mask X Range` column only accepts numerical input separated by commas, `0,1`, 0.1,2`...
- Open `Interfaces -> Data Manipulation -> Elwin`. Click on `Add Workspaces` and check that when adding data to the workspace dialog, the column for selecting indices only accepts integer numerical input separated by commas or dashes : `0,1,2-3`, `0-3`, `0,1`
-



*No release notes as it is a change with no effect to the user*



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
